search/ranker.cpp

```diff
@@ -608,4 +648,58 @@ void Ranker::ProcessSuggestions(vector<RankerResult> & vec) const
     ++i;
   }
 }
+
+string Ranker::GetLocalizedRegionInfoForResult(RankerResult const & result) const
+{
+  size_t const kMaxNumParts = 2;
+
+  auto const type = result.GetBestType(&m_params.m_preferredTypes);
+
+  storage::TCountryId id;
+  if (!result.GetCountryId(m_infoGetter, type, id))
+    return {};
+
+  vector<string> parts;
+  GetPathToRoot(id, m_countries, [&](storage::TCountryId const & id) {
+    parts.push_back(GetLocalizedCountryName(id));
+  });
+
+  if (parts.size() > kMaxNumParts)
+    parts.erase(parts.begin(), parts.end() - kMaxNumParts);
+
+  my::EraseIf(parts, [&](string const & s) { return s.empty(); });
+
+  if (!parts.empty())
+    return strings::JoinStrings(parts, ", ");
+
+  // Tries to get at least localized name for |id|, if |id| is a
+  // discussed territory.
+  auto name = GetLocalizedCountryName(id);
+  if (!name.empty())
+    return name;
+
+  // Tries to transform map name to the full name.
+  name = id;
+  storage::CountryInfo::FileName2FullName(name);
+  if (!name.empty())
+    return name;
+
+  return {};
```

Comment:
`return name` instead of 4 line above?

Reply:
Again, as above, I would like to keep this code as is, because today this method returns empty strings as a signal of invalid value, and this invalid return value is the same as invalid return value for FileName2FullName. But, imagine now, that this method returns boost optional. Current code still works, right? All what I need is to change return value type. But the code proposed by you don't work as expected.

Reply:
Ok.